misc/Kconfig

```diff
@@ -15,6 +15,7 @@ config HAS_FLASH_LOAD_OFFSET
 	This option is selected by targets having a FLASH_LOAD_OFFSET
 	and FLASH_LOAD_SIZE.
 
+if !HAS_DTS
```

Comment:
Will this break existing targets which have DTS and want to use flash, but don't define partitions defined yet? If so, could merging this patch be delayed until we've had time to fix those up?